apps/api/e2e/tests/competition.test.ts

```diff
@@ -514,6 +514,14 @@ describe("Competition API", () => {
     expect(Array.isArray(upcomingResponse.competitions)).toBe(true);
     expect(upcomingResponse.competitions.length).toBe(3);
 
+    // Verify metadata fields
+    expect(upcomingResponse.metadata).toBeDefined();
+    expect(upcomingResponse.metadata.total).toBe(3);
+    expect(upcomingResponse.metadata.limit).toBe(10); // default limit
+    expect(upcomingResponse.metadata.offset).toBe(0); // default offset
+    expect(typeof upcomingResponse.metadata.hasMore).toBe("boolean");
+    expect(upcomingResponse.metadata.hasMore).toBe(false); // 3 competitions < 10 limit
```

Comment:
The spec doesn't include the hasMore field. No problem if it's included, but seems good to keep that spec doc inline with what whatever we go with.

Reply:
I was referencing this issue: https://github.com/recallnet/js-recall/issues/470

Good call - we should ensure the spec doc and issues are always in sync